### PR TITLE
feat: allow filtering projects with operators

### PR DESCRIPTION
**Pull request by sjaanus**
This is first iteration. When we add more fields to be filterable with operators, we can have more reusable components for this.

### CODE REVIEW

File: src/lib/features/feature-search/feature-search-service.ts [hunk 1]
```diff
@@ -21,15 +25,48 @@ export class FeatureSearchService {
...
-            await this.featureStrategiesStore.searchFeatures({
-                ...params,
-                limit: params.limit,
-            });
+            await this.featureStrategiesStore.searchFeatures(
+                {
+                    ...params,
+                    limit: params.limit,
+                },
+                queryParams,
```
Comment (sjaanus) [inline src/lib/features/feature-search/feature-search-service.ts:35]: Currently expanding with new parameter, but eventually the top params will be removed.

File: src/lib/features/feature-search/feature-search-service.ts [hunk 2]
```diff
@@ -21,15 +25,48 @@ export class FeatureSearchService {
...
+            );
 
         return {
             features,
             total,
         };
     }
+
+    parseOperatorValue = (field: string, value: string): IQueryParam | null => {
```
Comment (sjaanus) [inline src/lib/features/feature-search/feature-search-service.ts:44]: Currently supporting just 4 operators.

File: src/lib/features/feature-toggle/feature-toggle-strategies-store.ts [hunk 3]
```diff
@@ -1033,5 +1036,27 @@ class FeatureStrategiesStore implements IFeatureStrategiesStore {
...
+            case 'IS':
+                query.where(param.field, '=', param.value);
+                break;
+            case 'IS_NOT':
+                query.where(param.field, '!=', param.value);
+                break;
+            case 'IS_ANY_OF':
+                query.whereIn(param.field, param.value as string[]);
+                break;
+            case 'IS_NOT_ANY_OF':
```
Comment (Tymek) [inline src/lib/features/feature-toggle/feature-toggle-strategies-store.ts:1054]: IS_NOT_IN would sound more natural to me. Or even `NOT_IN`. I think we should keep  domain language close to SQL. Same applies to the UI. What do you think @nicolaesocaciu ?

Comment (sjaanus) [reply src/lib/features/feature-toggle/feature-toggle-strategies-store.ts:1054]: I think we following language based on UI elements and those were taken 1 to 1 from linear. Maybe they have thought about it more than us.

Comment (sjaanus) [reply src/lib/features/feature-toggle/feature-toggle-strategies-store.ts:1054]: Either way we go, I think frontend should have same texts.

Comment (nicolaesocaciu) [reply src/lib/features/feature-toggle/feature-toggle-strategies-store.ts:1054]: I understand both expressions but for me it's a bit more clear when is see **"Is not any of"** compared with **"Is not in / Not in"**

I will ask also the team what they think, and we will have at some point some user testing with the filters and i will try to see what is the perception about these expressions

**Review by Tymek — APPROVED** [review]

**Pull request merged**